app/web/src/components/IdeHeader/IdeHeader.tsx

```diff
@@ -55,119 +51,37 @@ const TopButton = ({
   )
 }
 
-interface IdeHeaderProps {
-  handleRender: () => void
-  projectTitle?: string
-  projectOwner?: string
-  projectOwnerId?: string
-  projectOwnerImage?: string
-  projectId?: string
-  children?: ReactNode
-}
-
-const IdeHeader = ({
+export default function IdeHeader({
   handleRender,
-  projectOwner,
-  projectTitle,
-  projectOwnerImage,
-  projectId,
-  projectOwnerId,
-  children,
-}: IdeHeaderProps) => {
+  context,
+}: {
+  handleRender?: () => void
+  context: 'ide' | 'profile'
```

Comment:
this prop tells it where it's being used, which was the original inspiration for this refactor, see: https://github.com/Irev-Dev/cadhub/pull/533/files#r717442900